From 87351b819c8fce94c54c89b3695b7363e06ec727 Mon Sep 17 00:00:00 2001
From: Christophe Guibout <christophe.guibout@st.com>
Date: Wed, 10 Jul 2019 16:06:39 +0200
Subject: [PATCH] stm32mp1: add fastboot boot mode variable

Add new fastboot dedicated variable to allow getting back boot mode

Change-Id: I83843ee2b213f5a34cbfb23e00db595a5d7f365f
---
 arch/arm/mach-stm32mp/cpu.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm/mach-stm32mp/cpu.c b/arch/arm/mach-stm32mp/cpu.c
index 1fc3f2b89b..a6a7558632 100644
--- a/arch/arm/mach-stm32mp/cpu.c
+++ b/arch/arm/mach-stm32mp/cpu.c
@@ -407,6 +407,13 @@ static void setup_boot_mode(void)
 		env_set("boot_instance", cmd);
 #ifdef CONFIG_FASTBOOT_FLASH_MMC
 		env_set("fastboot.boot_instance", cmd);
+#ifdef CONFIG_STM32MP1_OPTEE
+		env_set("fastboot.boot_mode", "optee");
+#elif defined(CONFIG_STM32MP1_TRUSTED)
+		env_set("fastboot.boot_mode", "trusted");
+#else
+		env_set("fastboot.boot_mode", "basic");
+#endif
 #endif
 		break;
 	case BOOT_FLASH_NAND:
-- 
2.17.1

